specs/0108-community-health-dashboard.md

```diff
@@ -0,0 +1,49 @@
+- Feature Name: Community Health Dashboard
+- Created: 2020-10-23
+- Associated Bug: https://bugzilla.mozilla.org/show_bug.cgi?id=1672966
+
+# Summary
+
+Create a team-specific health dashboard targeted at Pontoon users with Manager rights.
+
+# Motivation
+
+Team managers as well as project managers should have the ability to understand how localization communities are performing. Team dashboards already provide some of the answers, including the overall status of completion and translation statistics, but the information is only available in the form of a current snapshot.
+
+There's no historic data to provide insights into how the community health is developing. Missing is the information on how completion and volume of unreviewed strings are changing over time. It's not clear what share of translations was copied from the Machinery. We also don't know the ratio of translations approved without a peer review.
+
+# Out of scope
+
+At this stage we’re only interested in presenting data we already collect in Pontoon. We're primarily focusing on exposing data to team managers; project managers will be able to obtain data across all teams if needed, e.g. by using scripts and the API.
+
+# Feature explanation
+
+A new "Insights" tab is made available on the Team page, positioned next to the "Contributors" tab. It consists of several sections, presenting data for the period of the last 12 months. Each section is described below. Fore more details about the design, see the [Mockup](#mockup) section.
+
+## Active users
+
+Pie charts showing ratios of active vs. all managers, reviewers and contributors. See the existing scripts ([1](https://github.com/flodolo/scripts/blob/954fa85/pontoon/active_contributors.py), [2](https://github.com/flodolo/scripts/blob/954fa85/pontoon/list_reviewers_with_contribution_stats.py)) for criteria of defining active users and for details on how to retrieve data.
+
+## Unreviewed suggestion lifespan
+
+A line chart showing time before suggestion gets approved or rejected. See the [existing script](https://github.com/flodolo/scripts/blob/954fa85/pontoon/unreviewed_suggestions_lifespan.py) for details on how to retrieve data. A tooltip showing the exact data at a given time appears when hovering over a chart.
+
+## Translation activity
+
+A combination of a line chart showing completion over time and a column chart showing approved translation submissions and source string additions over time in two separate columns. The translation submission column is a stack of two columns - human translations (`Translation.machinery_sources` not set) and Machinery translations (`Translation.machinery_sources` set). A tooltip showing the exact data at a given time appears when hovering over a chart.
+
+## Review activity
```

Comment:
I'm conflicted about the use of raw volume numbers instead of ratios. It feels like self-approval here is out of place, as the focus appears to be a depiction of what happens to unreviewed translations over time. As a community manager, I would be most interested in understanding of these unreviewed translations, our ratio of peer-approving translations to peer-rejecting translations is X, and this is what that means in terms of volume. Self-approval feels like it should be its own column maybe to compare peer review results against self-approval results?

Reply:
The charts actually show the ratios already:
- blue vs red: approvals vs rejections
- light blue vs dark blue: peer-approvals vs self-approvals

We can add the ratios in numeric form to the tooltip (in any combination, e.g. peer-approvals vs. (peer-)rejections and self-approvals vs. other peer review actions).

Reply:
I caught that, I should have been more clear. I'm wondering if a percentage ratio in the tooltip to reinforce the visual ratio is more helpful.